### ISSUE TITLE
Problem with webdirect on Edge browser

### ISSUE THREAD

**Issue by Catastrophi** [Don-Cryptus/tinyMCE#1]
Hello,

I tried to test this approach on filemaker server. 
An empty tinymce interface loads on Edge browser, without the data. The saving part works correctly though.  
I tried to access the file using Firefox (which is an unsupported browser according to Claris) and it seem to work correctly.
Any ideas on what could be the problem? 

Thanks.

**Comment by 0-don** [comment]
can you check the console in the web frame via right click

**Comment by Catastrophi** [comment]
`Mixed Content: The page at 'https://test.lan/fmi/webd/fm-tinymce%20-%20Copy' was loaded over HTTPS, but requested an insecure script 'http://langs/es.js'. This request has been blocked; the content must be served over HTTPS.
 Failed to load language: es `
This is what I get

**Comment by 0-don** [comment]
test it out now

**Comment by Catastrophi** [comment]
Hi, the language error disappeared, but still no data in the tinymce canvas. As before, it saves the data, but does not load it.

`Vaadin push loaded    vaadinPush.js?v=8.12.3.fmi:1 

[Violation] 'message' handler took 434ms    vaadinPush.js?v=8.12.3.fmi:1 

editor Initialized    VM235 b5550fd0-7287-46e0-a2c0-40b9a98636f9:1 `


Calling the setContent script by Clicking the button works and loads the content. Adding a small delay in the script by 1s doesn't seem to work

**Comment by Catastrophi** [comment]
It seems like the script is not called during tinymce initialization. 
I added a line to show a dialogue box before the setContnet script and it doesn't show up on tinymce initialization.

**Comment by 0-don** [comment]
you need to download the new version does it still have the same problem

**Comment by Catastrophi** [comment]
Hello, 
I did try with the new version and the problem persists.

Steps to reproduce: Download the fm-tinymce.fmp12 file from this repository.
Add password to admin user and activate fmwebdirect and fmurlscript previleges.
Upload to filemaker server and open using edge browser.

Result: Empty tinymce frame. On clicking the set content button, the content loads. In this case, the word "test".

Further experiment:
Close the file on the browser, open with filemaker pro and add a custom dialogue box in the setContent script to show a message.  
Open again with the browser. The dialogue box does not show and empty tinymce frame shows. Click on the set content button: The dialogue box pops up and the content gets loaded. 

This is what brought me to think that the script is not triggered on tinymce initialization.

**Comment by Catastrophi** [comment]
Did a bit more digging with the browser console. 

![imagen](https://user-images.githubusercontent.com/43213601/236150384-754e8a1c-d1ac-4ce4-9d7c-98f228888e80.png)

**Comment by 0-don** [comment]
which filemaker server version are you running

**Comment by Catastrophi** [comment]
FileMaker Server 19.6.3.302

**Comment by 0-don** [comment]
I removed the language defintions in the new version now try it out again and show me the bug from the iframe console is it still the same?

**Comment by 0-don** [comment]
btw 
![image](https://user-images.githubusercontent.com/70039285/236185571-ab2e1627-937a-4ec0-b40b-2f6430109624.png)
you need to allow this

**Comment by 0-don** [comment]
contact me via email don.cryptus@gmail.com

**Comment by Catastrophi** [comment]
> you need to allow this

This is already allowed in the file

Now, the following exceptions are caught from the moment the login credentials are entered.

DOMException: Failed to execute 'createEvent' on 'Document': The provided event type ('TouchEvent') is invalid.
Error: java.lang.IndexOutOfBoundsException: Index: 3, Size: 3
Error: java.lang.IndexOutOfBoundsException: Index: 3, Size: 3
TypeError: Array.prototype.forEach called on null or undefined
TypeError: Cannot convert undefined or null to object
TypeError: Cannot read properties of null (reading 'removeChild')
DOMException: Failed to set the 'outerHTML' property on 'Element': This element has no parent node.
ReferenceError: FileMaker is not defined

The DOMExceptions were there before. The java.lang error is new. This probably comes from the fact that the OS and browser is in Spanish. 

Not a good solution, but I could find an alternative to make it work: To call the setContent script from another script after login or opening the layout containing tinymce.

**Comment by 0-don** [comment]
yeah it has something to do with your browser language I disabled the language feature which should fix the problem but I dont understand why are you still getting it

**Comment by Catastrophi** [comment]
Hi, 
I think I figured out the problem. The browser language is one issue and adding the appropriate language file fixes it. 
The script not running is another problem. The cause is that we are trying to run the script before waiting for the previous scripts to be completed. I added a small delay before running the setContent script and it works now. Replace line 114 in the editor.js file with the following and recomplile.

`setTimeout(function(){ runFmScript('setContent', editor.getContent());}, 100);`

Thanks for the help and your time.
